identity/src/lib.rs

```diff
@@ -119,3 +119,5 @@ pub use error::{DecodingError, SigningError};
 pub use keypair::{Keypair, PublicKey};
 #[cfg(feature = "peerid")]
 pub use peer_id::{ParseError, PeerId};
+
+pub type KeyType = proto::KeyType;
```

Comment:
We need a way to tell users what type of key the `Keypair` is holding and that type seems to be a good choice because we do use it to represent different key types internally. Now thinking about it, the user won't put the `KeyType` back in so yeah, that should be kept private.

Reply:
> We need a way to tell users what type of key the `Keypair` is holding

If we want to have such a thing, we need to create our own enum to not be dependent on protobuf internals.